discord/ext/menus/__init__.py

```diff
@@ -978,45 +1066,52 @@ async def show_checked_page(self, page_number):
             # An error happened that can be handled, so ignore it.
             pass
 
-    async def show_current_page(self):
+    async def show_current_page(self) -> None:
         if self._source.is_paginating():
             await self.show_page(self.current_page)
 
-    def _skip_double_triangle_buttons(self):
+    def _skip_double_triangle_buttons(self) -> bool:
         max_pages = self._source.get_max_pages()
         if max_pages is None:
             return True
         return max_pages <= 2
 
-    @button('\N{BLACK LEFT-POINTING DOUBLE TRIANGLE WITH VERTICAL BAR}\ufe0f',
-            position=First(0), skip_if=_skip_double_triangle_buttons)
-    async def go_to_first_page(self, payload):
+    @button(
+        '\N{BLACK LEFT-POINTING DOUBLE TRIANGLE WITH VERTICAL BAR}\ufe0f',
+        position=First(0),
+        skip_if=_skip_double_triangle_buttons,
+    )
+    async def go_to_first_page(self, payload: discord.RawReactionActionEvent) -> None:
         """go to the first page"""
         await self.show_page(0)
 
     @button('\N{BLACK LEFT-POINTING TRIANGLE}\ufe0f', position=First(1))
-    async def go_to_previous_page(self, payload):
+    async def go_to_previous_page(self, payload: discord.RawReactionActionEvent) -> None:
         """go to the previous page"""
         await self.show_checked_page(self.current_page - 1)
 
     @button('\N{BLACK RIGHT-POINTING TRIANGLE}\ufe0f', position=Last(0))
-    async def go_to_next_page(self, payload):
+    async def go_to_next_page(self, payload: discord.RawReactionActionEvent) -> None:
         """go to the next page"""
         await self.show_checked_page(self.current_page + 1)
 
-    @button('\N{BLACK RIGHT-POINTING DOUBLE TRIANGLE WITH VERTICAL BAR}\ufe0f',
-            position=Last(1), skip_if=_skip_double_triangle_buttons)
-    async def go_to_last_page(self, payload):
+    @button(
+        '\N{BLACK RIGHT-POINTING DOUBLE TRIANGLE WITH VERTICAL BAR}\ufe0f',
+        position=Last(1),
+        skip_if=_skip_double_triangle_buttons,
+    )
+    async def go_to_last_page(self, payload: discord.RawReactionActionEvent) -> None:
         """go to the last page"""
         # The call here is safe because it's guarded by skip_if
         await self.show_page(self._source.get_max_pages() - 1)
 
     @button('\N{BLACK SQUARE FOR STOP}\ufe0f', position=Last(2))
-    async def stop_pages(self, payload):
+    async def stop_pages(self, payload: discord.RawReactionActionEvent) -> None:
         """stops the pagination session."""
         self.stop()
 
-class ListPageSource(PageSource):
+
+class ListPageSource(PageSource[T], Generic[T]):
```

Comment:
This doesn't need to re-inherit Generic[T]
```suggestion
class ListPageSource(PageSource[T]):
```